frappe/desk/listview.py

```diff
@@ -65,14 +65,33 @@ def get_group_by_count(doctype: str, current_filters: str, field: str) -> list[d
 	if not meta.has_field(field) and not is_default_field(field):
 		raise ValueError("Field does not belong to doctype")
 
-	data = frappe.get_list(
-		doctype,
-		filters=current_filters,
-		group_by=f"`tab{doctype}`.{field}",
-		fields=["count(*) as count", f"`{field}` as name"],
-		order_by="count desc",
-		limit=50,
-	)
+	if field == "owner":
+		data = frappe.get_list(
+			doctype,
+			filters=[*current_filters, {field: ["!=", frappe.session.user]}],
+			group_by=f"`tab{doctype}`.{field}",
+			fields=["count(*) as count", f"`{field}` as name"],
+			order_by="count desc",
+			limit=50,
+		)
+
+		data.extend(
+			frappe.get_list(
+				doctype,
+				filters=[*current_filters, {field: frappe.session.user}],
+				group_by=f"`tab{doctype}`.{field}",
+				fields=["count(*) as count", f"`{field}` as name"],
+			)
+		)
```

Comment:
So, I feel it should be kinda doable. Since the query is first building a complete list of all owners + their count. The `limit 50` clause is only avoiding sending of that data. 

Maybe just fetch all the data and send the first 50 + self. This is so counter-intuitive but it might just work.

Reply:
(open to alternate ideas)

Reply:
Eh possibly even simpler: `order_by owner = self desc, count desc limit 50` :smile: